store/sqlstore/store.go

```diff
@@ -920,7 +920,11 @@ func (ss *SqlStore) AlterColumnTypeIfExists(tableName string, columnName string,
 	}
 
 	if err != nil {
-		mlog.Critical("Failed to alter column type", mlog.Err(err))
+		msg := "Failed to alter column type."
+		if mySqlColType == "JSON" && postgresColType == "jsonb" {
+			msg += " It is likely you might have invalid JSON values in the column. Please fix the values manually and run the migration again."
+		}
+		mlog.Critical(msg, mlog.Err(err))
```

Comment:
Gentle reminder to check our logging [principles](https://developers.mattermost.com/contribute/server/style-guide/#log-levels) before merging this change.